### DISCUSSION TITLE
Performous 1.2.0-rc1

### DISCUSSION THREAD

**Discussion by Baklap4** [performous/performous#685] [Announcements]
It's been quite some time we've made a release (almost 6 years). Within this release we've mainly stabilized Performous and focused on automation and solving bugs.
However there are also a few new features within this release!

# Upcoming releases

We've made some big gains in automation. This is why we're going to change the releasing strategy.
With the upcoming release we'll have a release candidate. A release candidate will be a branch on its own (with a tag) where only hotfixes will be applied to.
After a succesful testing period we'll then make a release. If the testing period is not succesful we'll create a new release candidate and repeat the testing.
Once a release candidate is ready we'll make a release branch and tag that branch as the official release.

We'll try to create more releases in less time. Preferably after every PR. As we always did we will use semver as version strategy.

# New features

- Webserver overhaul (@baklap4)
- Add song paths in game (@nieknooijens)
- Configurable playlist duration (@nieknooijens)
- Audio system overhaul (@Lord-Kamina)
- Added Pro Mode to the drums (@earlye)
- Metadata caching (@baklap4)
- Changing languages from within the menu (@baklap4)
- Windows installer (@baklap4)
- Improved frequency detection while singing (@Tronic)
- Note graph scaling: Static, dynamic (@twollgam)
- Difficulty levels for singing (@isadofschi)
- Proper support for Hi-DPI displays (@Lord-Kamina)
- Auto-merging of duet tracks (@Lord-Kamina)

# Bugfixes

- Remembering sort order (@nieknooijens)
- Fullscreen mode on MacOS (@Tronic)
- Fix a crash in playlist screen when there was an empty playlist (@nieknooijens)
- Several memory leaks (@OznOg)
- Shortcuts in game representation (@baklap4)
- Path screen selection on top (@OznOg)
- Devices were off by one (@OznOg)
- Missing cache entries (@OznOg)
- Autodetecting webcam (@baklap4)
- Pitchwave order is now based on score. Highest gets shown on top (@baklap4)
- Fix a crash where a player gets more than 10k points (@baklap4)
- Nav keys getting stuck (@Tronic)
- Fixed a couple missing glyphs in the `Goblin One` font (@Lord-Kamina)
- Updated version of `Droid Sans` font (@Lord-Kamina)
- Default aspect ratio is now 16:9 (@Lord-Kamina)

# Translations

- Dutch (@baklap4 & @nieknooijens)
- German (@twollgam)
- French (@OznOg)
- Spanish (@Lord-Kamina)
- Hungarian (@skyli91)
- Slovak (@jose1711)
- Japanese (@nieknooijens)

# Other

- using c++ std more (@Tronic & @OznOg)
- Use libxml++3 or libxml++5 when available (@yoda-jm & @baklap4)
- Better unicode handling (@Lord-Kamina)
- Change from OpenGL 2.1 to OpenGL 3.3 and fixed display issues on older intel GPUs (@Tronic, @Lord-Kamina)
- CI + Artifacts (@baklap4, @Lord-Kamina, @deuteragenie, @ooshlablu)
- Native compilation on Windows (@baklap4)
- Lots of cmake fixes and cleanup (@OznOg, @Lord-Kamina, @yoda, @baklap4)
- Refactoring (@twollgam)
- Meson build support (@Tronic, @deuteragenie)
- Using libglm instead of in-house math. (@Lord-Kamina)
- Various fixes in the macOS bundling script (and better support for Homebrew) (@Lord-Kamina, @deuteragenie)

# Notes

On Windows you might want to enable UTF-8 support throughout the system. This way windows can support special characters within song paths. You can enable this by doing this:
1. Windows btn + R -> `intl.cpl`
2. Hit the 2nd tab called "administritive"
3. Hit the button "Change System Locale..."
4. Tick the checkbox "Beta: Use Unicode UTF-8 for worldwide language support"
5. Restart pc

<hr /><em>This discussion was created from the release <a href='https://github.com/performous/performous/releases/tag/1.2.0-rc1'>Performous 1.2.0-rc1</a>.</em>

**Comment by Baklap4** [comment]
## Testing period

We'll be testing until the **12th of october**

Meanwhile we will gather your feedback within this thread.
From this feedback we'll decide what to fix and what not to fix for the official 1.2.0 release.

I'll be updating this thread with a list of things we wanna get fixed.

### Finetune/fixup for official 1.2.0 release

- [ ] ~~Windows Installer allow non-admins to install it too~~ (https://github.com/performous/performous/issues/689)
    - [ ] ~~`set(CPACK_NSIS_DEFINES "RequestExecutionLevel user")`~~
- [x] Fix "Auto" language detection (https://github.com/performous/performous/pull/688)
- [x] Add Slovak to the ingame translations / check for missing languages (https://github.com/performous/performous/pull/688)
- [ ] ~~Fix the pango warning: Couldn't load font droid sans mono bolt not rotated"~~ (https://github.com/performous/performous/issues/690)
    - [ ] ~~Probably use the correct full name~~
- [x] Fix Aubio alias (https://github.com/performous/performous/pull/688)
- [x] ~~Write blog post for final version to be submitted for final release within the https://github.com/performous/web project~~ (https://github.com/performous/web/pull/12)
- [x] ~~Look into signing the executables. If not easily done submit the windows installer to microsoft for manual scanning~~
    - [ ] For now we'll submit the installer to microsoft.

**Reply by twollgam** [reply]
How is the state of the release? The planned release date of 12th october passed. Which tasks are still open? And can I support anyone finishing open tasks?

**Comment by diver66** [comment]
Hello,
can somebody pls help me out. How do I bump my version 1.1-1140-g9c66e273 to 1.2 (running on raspberry pi 4 / Linux / Dietpi)

**Reply by Lord-Kamina** [reply]
Well that depends... How'd you get it in the first place? 😂

I wasn't aware it worked properly on any distros for RPi.

**Reply by diver66** [reply]
hi, yes it does work -  for the installation I just followed the ubuntu debian installation manual  :-)

**Reply by Baklap4** [reply]
As stated on the previous release discussion, there's a branch called 1.2.0-rc1. A tag has been made on this branch. Do note this is a pre-release. Once we merge this branch back to master  we'll tag the official 1.2.0 version (should be around end of october). Once on that tag there's no need to bump the version as that's been done there already. I think after 1.2.0 we'll have to look into a good version/branching strategy. Some discussion already found place in discord.

**Reply by Tronic** [reply]
Could just do a `git pull` on your existing repository (I assume you installed from git master branch) to update it and then follow the original guide to do a recompile. Or if you deleted your old copy, just do the same install instructions again and you should get the latest development version.
